server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridTiler.java

```diff
@@ -259,8 +268,8 @@ protected int setValuesByRasterization(int xTile, int yTile, int zTile, GeoShape
             return valuesIndex;
         }
 
-        private int setValuesForFullyContainedTile(int xTile, int yTile, int zTile,
-                                                   GeoShapeCellValues values, int valuesIndex, int targetPrecision) {
+        private int setValuesForFullyContainedTile(int xTile, int yTile, int zTile, CellValues values, int valuesIndex,
+                                                   int targetPrecision) {
```

Comment:
Don't we need to filter out  the tiles that do not intersect with the provided bounds (if provided)